Mathlib/GroupTheory/Exponent.lean

```diff
@@ -386,3 +387,51 @@ theorem card_dvd_exponent_pow_rank' {n : ℕ} (hG : ∀ g : G, g ^ n = 1) :
 #align card_dvd_exponent_nsmul_rank' card_dvd_exponent_nsmul_rank'
 
 end CommGroup
+
+section PiProd
+
+open Finset Monoid
+
+@[to_additive]
+theorem Monoid.exponent_pi_zero {ι : Type*} {M : ι → Type*} [∀ i, Monoid (M i)] {j : ι}
+    (hj : exponent (M j) = 0) : exponent ((i : ι) → M i) = 0 := by
+  rw [@exponent_eq_zero_iff, ExponentExists] at hj ⊢
```

Comment:
Is this true as an `Iff` with an existential?

Reply:
(and is the `Prod` version too, with an `or`?)

Reply:
Note: the version I provided here does *not* assume `Fintype ι` as `Monoid.exponent_pi` does. Without that, the `↔` does not hold (consider the product `Π n, ZMod n` for instance).

With the `Fintype ι` condition, then yes, it's true, but it follows immediately from `Finset.lcm_eq_zero_iff`, and I would expect users to use that if they want this result. Similarly, the prod version holds, but follows immediately from `lcm_eq_zero_iff` (once I switch from `Nat.lcm` to `lcm`, which I just realized I should have done).

So, these are the lemmas, but I propose *not* including them:
```lean
@[to_additive]
lemma Monoid.exponent_pi_zero_iff {ι : Type*} [Fintype ι] {M : ι → Type*} [∀ i, Monoid (M i)] :
    exponent ((i : ι) → M i) = 0 ↔ ∃ i, exponent (M i) = 0 := by
  rw [Monoid.exponent_pi, Finset.lcm_eq_zero_iff]
  simp

@[to_additive]
theorem Monoid.exponent_prod_zero_iff {M₁ M₂ : Type*} [Monoid M₁] [Monoid M₂] :
    exponent (M₁ × M₂) = 0 ↔ (exponent M₁ = 0) ∨ (exponent M₂ = 0) := by
  rw [exponent_prod, lcm_eq_zero_iff]
```